app/helpers/validation/profile.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+import logging
+
+import geojson
+from flask import abort
+from shapely.geometry import shape
+
+from app import make_error_msg
+from app.helpers.profile_helpers import PROFILE_MAX_AMOUNT_POINTS, PROFILE_DEFAULT_AMOUNT_POINTS
+from app.helpers.validation import srs_guesser
+
+logger = logging.getLogger(__name__)
+
+
+def read_linestring(request_object):
+    # param geom, list of coordinates defining the line on which we want a profile
+    linestring = None
+    if 'geom' in request_object.args:
+        linestring = request_object.args.get('geom')
+    elif request_object.is_json and len(request_object.get_data(as_text=True)) > 0:
+        linestring = request_object.get_json()
+    if not linestring:
+        abort(400, "No 'geom' given, cannot create a profile without coordinates")
+    try:
+        geom = geojson.loads(linestring, object_hook=geojson.GeoJSON.to_instance)
+    except ValueError:
+        raise abort(400, "Error loading geometry in JSON string")
```

Comment:
Remove the `raise` and should not be a 500 here ? Added the value of ValueError in a log message would be good. Maybe also in the abort message ?

Reply:
I think it is not a 500, it will depend on the user's input (if it's a malformed GeoJSON but a valid JSON)